src/DIRAC/TransformationSystem/Service/TransformationManagerHandler.py

```diff
@@ -168,21 +201,26 @@ def export_getTransformations(
 
     types_getTransformation = [[int, str]]
 
-    @classmethod
-    def export_getTransformation(cls, transName, extraParams=False):
-        return cls.transformationDB.getTransformation(transName, extraParams=extraParams)
+    def export_getTransformation(self, transName, extraParams=False):
+        # check first if transformation exists to avoid returning permissions error for non-existing transformation
```

Comment:
If you "walrus" checkPermissions, you don't need this special case.

Also this is "getTransformation", so I pointed out similar functions, where checkPermissions was not added. So I am not sure about where you want to add checkPermissions and where not.

Reply:
We added it everywhere, where we can identify the transformation, i.e. where "transName" is accessible. Seems good enough for me.

Reply:
I definitely shouldn't have cut-and-paste the walrus code.

Reply:
> I definitely shouldn't have cut-and-paste the walrus code.

:innocent: